org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/Variable.java

```diff
@@ -75,21 +75,13 @@ public Variable(String name, Value value) {
     }
 
     /**
-     * Get the addition information of this variable.
+     * Get the declaring type of this variable if it is a field declared by some class.
      *
-     * @return the addition information of this variable.
+     * @return the declaring type of this variable.
      */
-    public String getNameDifferentiator() {
-        if (this.local != null) {
-            return "Local";
-        }
-
-        if (this.argumentIndex >= 0) {
-            return "Argument";
-        }
-
-        if (this.field != null && this.field.declaringType() != null) {
-            return this.field.declaringType().name();
+    public Type getDeclaringType() {
```

Comment:
a lot of places need to get the declare type of this variable, create an getter method to avoid duplicate code.